src/runtime_src/core/tools/common/ReportCu.cpp

```diff
@@ -160,22 +177,40 @@ populate_cus_new(const xrt_core::device *device)
     pt.push_back(std::make_pair("", ptCu));
   }
 
+  std::vector<ps_kernel_data> psKernels;
+  if (getPSKernels(psKernels, device) < 0) {
+      std::cout << "WARNING: 'ps_kernel' invalid. Has the PS kernel been loaded? See 'xbutil program'.\n";
+      return pt;
+  }
+
+  uint32_t psk_inst = 0;
+  uint32_t num_scu = 0;
   boost::property_tree::ptree pscu_list;
   for (auto& stat : scu_stats) {
     boost::property_tree::ptree ptCu;
-    std::string scu_name = stat.name;
-    auto found = scu_name.rfind("scu");
-    if (found > 0) {
-        std::string scu_i = scu_name.substr(found + 3);
-        scu_name = scu_name.substr(0, found - 1);
-        scu_name.append(scu_i);
+    std::string scu_name = "Illegal";
+    if (psk_inst >= psKernels.size()) {
+      scu_name = stat.name; //This means something is wrong
+    } else {
+      scu_name = psKernels.at(psk_inst).pkd_sym_name;
+      scu_name.append("_");
+      scu_name.append(std::to_string(num_scu));

```

Comment:
It's hard to understand what is happening here, can you add a comment to provide an example of the before and after scu_name would look like

Reply:
Done